### DISCUSSION TITLE
json_encode in ComponentChecksumManager should enforce object conversion

### DISCUSSION THREAD

**Discussion by 16hands** [livewire/livewire#3881] [7. Bugs (archived)]
Hi 

We ran into an odd issue with livewire, it was fine on plesk servers but not our production HA cluster, we consistently got the 'Livewire encountered corrupt data when trying to hydrate ... ' error 

With some digging we found the hash_hmac function was hashing very slightly different JSON, the checksum would generate JSON with empty arrays like this (from a much larger JSON packet):
`"filter_tags":[[],[],[],[],[]]} `

and the verification was generating JSON like this:
`"filter_tags":[{},{},{},{},{}]}`

I'm not a JS guru, but my gut is using the 'everything is an object in JS' kind of thinking and this makes sense that it would come back as empty objects, but its odd that the same version of php (different package maintainers) would come back with different JSON, its not so much that it failed, its that it was different between servers with the same php version 

Would you consider setting JSON encode in ComponentChecksumManager, with the  'JSON_FORCE_OBJECT' flag?,  it ensures its comparing the same encoding rather than relaying on the default PHP settings which can be distro bound 

```
$stringForHashing = ''
       .json_encode($fingerprint, JSON_FORCE_OBJECT)
       .json_encode($memoSansChildren, JSON_FORCE_OBJECT);
```

Thank you for a wonderful project

Kind regards
Carl

**Comment by joshhanley** [comment]
@16hands I think that could solve some issues we also have with javascript re-sorting arrays, but may have unintended side effects for everyones code that is now expecting javascript arrays in the front end instead of objects.

As such this would probably be a breaking change.

But I do think it's something to potentially explore.
